bin/fpdiff.py

```diff
@@ -80,6 +80,12 @@ def fpdiff_helper(filename1,filename2,relative_error,small,
  # handle the error and so we shouldn't do anything weird here).
  tmpfile1 = read_file(filename1)
  tmpfile2 = read_file(filename2)
+ # this line catches the error when an empty selftest reference data file is 
+ # provided and will return a FAIL (e.g. wrong filename is passed for the selftest, etc.)
```

Comment:
Ah, one final thing: the comment isn't entirely correct -- if the wrong filename is passed, `readfile(...)` will try to open a file that doesn't exist and will throw an error. In this case, the user won't even reach this line. If you remove the text "(e.g. wrong filename is passed for the selftest, etc.)", I'm happy to merge this in

Reply:
done